flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -20,6 +20,7 @@
 
 import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.state.CheckpointListener;
```

Comment:
nit: Considering that you have anyways backported the CheckpointListener under the current package, wondering if you need to migrate all the usages as part of this change? Would it not be easy to just do the refactoring in a separate PR?

Reply:
I'm not sure about what's being followed as part of the Flink project. If this is the convention, then it definitely makes sense to put it here. My only issue with combining the refactor along with the feature changes was that the diff was getting harder to review due to the number of files touched.

Reply:
I think the convention is that the refactoring should be in a separate commit. So the commit history won't be messed up. Here the refactoring is put in [this commit](https://github.com/apache/flink/pull/13784/commits/d9b4a7405663488e53da1af82e482e9e0a809618).